blocks/bricks/__init__.py

```diff
@@ -345,15 +345,18 @@ def __init__(self, input_dim, output_dim, num_pieces, **kwargs):
         self.output_dim = output_dim
         self.num_pieces = num_pieces
 
-        self.linear_transformation = Linear(
-            name=self.name + '_linear_to_maxout', input_dim=input_dim,
-            output_dim=output_dim * num_pieces, weights_init=self.weights_init,
-            biases_init=self.biases_init, use_bias=self.use_bias)
-        self.maxout_transformation = Maxout(name=self.name + '_maxout',
-                                            num_pieces=num_pieces)
+        self.linear_transformation = Linear(name=self.name +
```

Comment:
I would stick to doing all configuration in `_push_allocation_config` for the purpose of simplicity and leave the current MLP brick as is. I am just scared of all this variety of possible behaviours we will have to deal with.

But when you create your children in the constructor as here, I think it is okay to pass the constructor arguments downwards. What I would not like is to configure the created child in the constructor. That would overlap with `_push_allocation_config`